rx-java/rxgrpc-stub/src/main/java/com/salesforce/rxgrpc/stub/Queues.java

```diff
@@ -0,0 +1,598 @@
+/*
+ *  Copyright (c) 2019, salesforce.com, inc.
+ *  All rights reserved.
+ *  Licensed under the BSD 3-Clause license.
+ *  For full license text, see LICENSE.txt file in the repo root  or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/*
+ * Copyright (c) 2011-2017 Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.salesforce.rxgrpc.stub;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Queue;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.salesforce.reactivegrpc.common.Supplier;
+import io.reactivex.exceptions.Exceptions;
+import io.reactivex.internal.fuseable.SimplePlainQueue;
+import io.reactivex.internal.fuseable.SimpleQueue;
+import io.reactivex.internal.queue.SpscArrayQueue;
+import io.reactivex.internal.queue.SpscLinkedArrayQueue;
+
+/**
+ * Queue utilities and suppliers for 1-producer/1-consumer ready queues adapted for
+ * various given capacities.
+ */
+public final class Queues {
```

Comment:
consider using `JCTools` queues or, if possible, RxJava2's ? (but I think they're internal only), rather than copying the whole implementations from Reactor. At the very least, I don't think the system properties make much sense...